### PR TITLE
Azure::*::_detail::Version => Azure::*::PackageVersion

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -15,7 +15,7 @@
 - Renamed `NoRevoke` to `EnableCertificateRevocationListCheck` for `Azure::Core::Http::CurlTransportSSLOptions`.
 - Renamed `GetString()` to `ToString()` in `Azure::Core::DateTime`.
 - Renamed `GetUuidString()` to `ToString()` in `Azure::Core::Uuid`.
-- Moved `Version` from `Azure::Core::Details` to `Azure::Core`.
+- Renamed `Azure::Core::Details::Version` to `Azure::Core::PackageVersion`.
```
[the remark this answers is not in the document]
Comment (antkmsft) [reply sdk/core/azure-core/CHANGELOG.md:18]: This is an exception, as @RickWinter said. Customers were using it as if it was public, and knew it under that name in that namespace.

File: sdk/storage/azure-storage-common/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## 12.0.0-beta.9 (Unreleased)
 
+### Breaking Changes
+
+- Renamed `Azure::Storage::Common::Details::Version` to `Azure::Storage::Common::PackageVersion`.
```
[the remark this answers is not in the document]
Comment (antkmsft) [reply sdk/storage/azure-storage-common/CHANGELOG.md:7]: I explained above.

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 - DataLake client constructors won't automatically convert blob url to dfs url anymore.
 - String conversion functions of extensible enums were renamed from `Get()` to `ToString()`.
 - Moved `SecondaryHostForRetryReads` out of retry options, now it's under `DataLakeClientOptions`.
+- Renamed `Azure::Storage::Files::DataLake::Details::Version` to `Azure::Storage::Files::DataLake::PackageVersion`.
```
[the remark this answers is not in the document]
Comment (antkmsft) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:14]: I explained above.

File: sdk/core/azure-core/CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahsonkhan) [reply sdk/core/azure-core/CHANGELOG.md:18]: I don't think we have any evidence that customers were using it, and even if they were, this sets an incorrect precedence that we consider changes to internal types as "breaking", which we don't.

I can fix this separately later, but I think this should not be under breaking changes, but rather new features.

File: sdk/core/azure-core/inc/azure/core/version.hpp [hunk 4]
```diff
@@ -22,7 +22,7 @@ namespace Azure { namespace Core {
   /**
    * @brief Provides version information.
    */
-  class Version {
+  class PackageVersion {
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/version.hpp:25]: Reminder to rename the header files too, later once all the namespace moves are done.

Comment (antkmsft) [reply sdk/core/azure-core/inc/azure/core/version.hpp:25]: Sure. But we will be doing all the file movements later, in separate PR(s). Please ping me if you expect to be fixed in this PR.

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core/inc/azure/core/version.hpp (hunk 4 above)
Comment (ahsonkhan) [reply sdk/core/azure-core/inc/azure/core/version.hpp:25]: Separate is good.

File: sdk/core/azure-core/CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (RickWinter) [reply sdk/core/azure-core/CHANGELOG.md:18]: I agree that it is technically new.  And its true that it was being used so its breaking (albeit moving _details stuff to public is not breaking).  Either would place in the Changelog is fine.  
My belief is customers don't look heavily at the Changelog other than to see quickly what is going on, or to see if something was in the new update that may have caused their issue.  As long as we call it out, I don't think anyone is going to nitpick on which section it was under.

File: sdk/core/azure-core/inc/azure/core/version.hpp (hunk 4 above)
Comment (RickWinter) [reply sdk/core/azure-core/inc/azure/core/version.hpp:25]: There are lots of header renames we need to do.  We will do them after the types have final homes.

**Pull request merged**